### ISSUE TITLE
Feature request - Add majority output_type

### ISSUE THREAD

**Issue by Brent-Edwards** [PDAL/PDAL#3468]
It would be really useful to add a "majority" or "mode" output_type to compute majority class rasters.

Ref - [https://pdal.io/stages/writers.gdal.html](url)

**Comment by hobu** [comment]
Considerations:

* Adding this would preclude using stream mode, which is currently possible with the statistics provided
* What is the binning strategy to be for a continuous data type (what would mode of GPStime be?)
* What is the data type of the mode band? If it is the same as the `dimension` data type, that could be a complication for downstream consumers.

**Comment by Brent-Edwards** [comment]
> Considerations:
> 
> * Adding this would preclude using stream mode, which is currently possible with the statistics provided
> * What is the binning strategy to be for a continuous data type (what would mode of GPStime be?)
> * What is the data type of the mode band? If it is the same as the `dimension` data type, that could be a complication for downstream consumers.

I agree that the majority or mode would be useful for discrete data types like classification. The data type could be user defined or computed on the fly to ensure that it can hold the output value - ex. 8bit unsigned for classification would be enough to cover standard and extended classes.
